.github/workflows/ci.yml

```diff
@@ -118,6 +118,33 @@ jobs:
           $MAVEN ${MAVEN_TEST} -T C1 clean verify -DskipTests -P gib,errorprone-compiler \
             -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
 
+  deprecation-checks:
```

Comment:
This is going to increase the build time, notice that we don't have many runners for CI here and requesting another job may cause that waiting time for CI could increase.
We already compile everything in maven-checks so maybe we could add a step maven-checks job?